### PR TITLE
Various Game Improvements

### PR DESCRIPTION
**Pull request by Vinyzu**
- Implemented Savestates MongoDB
- Implemented States (for Savestates)
- Renamed Points to Coins

### CODE REVIEW

File: src/sincere_singularities/data/savestates.py [hunk 1]
```diff
@@ -53,25 +73,33 @@ def add_many_user_states(self, datas: Iterable[StateFormat]) -> None:
...
             self.add_user_state(data)
 
-    def save_game_state(self, player_id: str, state: dict[str, Any]) -> None:
+    def save_game_state(self, player_id: int, state: State) -> None:
         """Save state
 
         Args:
-            player_id (str): User id
+            player_id (int): User id
             state (str): State
```
Comment (koviubi56) [inline src/sincere_singularities/data/savestates.py]:
```suggestion
            state (State): State
```

**Review by koviubi56 — CHANGES_REQUESTED** [review]
Just fix that, then it's good.

**Review by koviubi56 — APPROVED** [review]
LGTM!

**Comment by WassCodeur** [comment]
Great

**Pull request merged**